src/memray/_memray/record_writer.cpp

```diff
@@ -45,10 +45,19 @@ RecordWriter::RecordWriter(
             d_stats,
             command_line,
             ::getpid(),
+            0,
+            0,
             getPythonAllocator()};
     strncpy(d_header.magic, MAGIC, sizeof(d_header.magic));
 }
 
+void
+RecordWriter::setMainTidSkip(thread_id_t main_tid, size_t skipped_frames_on_main_tid)
```

Comment:
Sorry for the nit but i found the name a bit odd

```suggestion
RecordWriter::setMainTidAndSkipFrames(thread_id_t main_tid, size_t skipped_frames_on_main_tid)
```